prov/verbs/src/ep_rdm/verbs_rdm_cm.c

```diff
@@ -615,21 +617,76 @@ ssize_t fi_ibv_rdm_conn_cleanup(struct fi_ibv_rdm_conn *conn)
 	return ret;
 }
 
+static int fi_ibv_rdm_poll_cq(struct fi_ibv_rdm_ep *ep)
```

Comment:
Maybe call here common poll function instead this one and skip unneeded logic basing on endpoint or connection state? Doing like this, you may save inlining on hot path paying cost of couple checks. Anyway it's up to you ;)